qiskit/transpiler/passes/optimization/commutation_analysis.py

```diff
@@ -111,15 +112,21 @@ def _commute(node1, node2, cache):
 
     node1_key = (node1.op.name, str(node1.op.params), str(qarg1))
     node2_key = (node2.op.name, str(node2.op.params), str(qarg2))
+
+    op1 = Operator(node1.op)
```

Comment:
Do we actually need the Operator class here or can we just work with the raw ndarray? There can often be a speedup if we work directly with numpy directly